pkg/utils/release/factory.go

```diff
@@ -0,0 +1,43 @@
+package release
+
+import kubeCl "github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes"
+
+// Contains all embebed interfaces for release operations.
+type ReleaseService interface {
+	// Interface for all component related operations.
+	ComponentsInterface
+
+	// Interface for all pipeline related operations.
+	PipelinesInterface
+
+	// Interface for all plan related operations.
+	PlansInterface
+
+	// Interface for all release related operations.
+	ReleasesInterface
+
+	// Interface for all sbom related operations.
+	SbomInterface
+
+	// Interface for all secret related operations.
+	SecretsInterface
+
+	// Interface for all snapshot related operations.
+	SnapshotsInterface
+
+	// Interface for all strategy related operations.
+	StrategiesInterface
+}
+
+// Factory to initialize the comunication against different API like github or kubernetes.
+type releaseFactory struct {
```

Comment:
I don't really understand what is going on in this file? Again, cause I don't have a lot of GoLang expertise. But NewSuiteController is meant to return a ReleaseService interface but we are returning a releaseFactory how is golang allowing that? Is it cause of the functions in the interfaces are assigned to the factory class?